Refactor bootstrap.js for E2E.

ankitrox opened the pull request:
## Summary

<!-- Please reference the issue this PR addresses in the following list. -->
Addresses issue:

- #10369 

## Relevant technical choices

<!-- Please describe your changes. -->

## PR Author Checklist

- [ ] My code is tested and passes existing unit tests.
- [ ] My code has an appropriate set of unit tests which all pass.
- [ ] My code is backward-compatible with WordPress 5.2 and PHP 7.4.
- [ ] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [x] My code has proper inline documentation.
- [x] I have added a QA Brief on the issue linked above.
- [x] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Code Reviewer Checklist

- [ ] Run the code.
- [ ] Ensure the acceptance criteria are satisfied.
- [ ] Reassess the implementation with the IB.
- [ ] Ensure no unrelated changes are included.
- [ ] Ensure CI checks pass.
- [ ] Check Storybook where applicable.
- [ ] Ensure there is a QA Brief.
- [ ] Ensure there are no unexpected significant changes to file sizes.

## Merge Reviewer Checklist

- [ ] Ensure the PR has the correct target branch.
- [ ] Double-check that the PR is okay to be merged.
- [ ] Ensure the corresponding issue has a ZenHub release assigned.
- [ ] Add a changelog message to the issue.

techanvil commented on tests/e2e/config/bootstrap.js:
```diff
@@ -144,108 +243,27 @@ function observeConsoleLogging() {
-		// At this point, any unexpected message will result in a test failure.
-
-		// Check if it's raised by the AMP plugin.
+		// Check if it's raised by the AMP plugin
```
Please restore the removed/modified comment here.

techanvil reviewed (changes requested):
Thanks @ankitrox, this is looking good - it just needs a number of inline comments to be restored or fixed. Please see my comments on the PR for details.

ankitrox commented:
Thank you @techanvil for looking into the code review. I have addressed the comments regarding comments restoration and also restored at some other places where those were missing.

Assigning this to you to review the changes.

Thanks again!

techanvil commented on tests/e2e/config/bootstrap.js:
```diff
@@ -66,6 +64,116 @@ const OBSERVED_CONSOLE_MESSAGE_TYPES = {
+		pattern:
+			'Failed to load resource: the server responded with a status of',
+	},
+	// WordPress font loading warnings
```
Comments should end in a full stop. I'll apply this change myself to save the round trip.

```suggestion
	// WordPress font loading warnings.
```
